src/channel-header.tsx

```diff
@@ -1,40 +1,84 @@
 // tslint:disable-next-line:no-unused-variable
 import * as React from 'react';
 import AppBar from 'material-ui/AppBar';
+import { Tab } from 'material-ui/Tabs';
 
 import { Action } from './lib/action';
-import { ChannelBase } from './lib/models/api-shapes';
+import { Channel, ChannelBase } from './lib/models/api-shapes';
+import { ChannelListViewModel } from './channel-list';
 import { Model, fromObservable } from './lib/model';
+import { isChannel } from './channel-utils';
 import { SimpleView } from './lib/view';
 import { Store } from './lib/store';
+import { when } from './lib/when';
 
 export class ChannelHeaderViewModel extends Model {
-  toggleDrawer: Action<boolean>;
-  @fromObservable isDrawerOpen: boolean;
   @fromObservable selectedChannel: ChannelBase;
+  @fromObservable channelInfo: Channel;
+  @fromObservable members: Array<string>;
+  @fromObservable topic: string;
+  @fromObservable isDrawerOpen: boolean;
+  toggleDrawer: Action<boolean>;
 
-  constructor(readonly store: Store) {
+  constructor(public readonly store: Store, listViewModel: ChannelListViewModel) {
     super();
 
     let isDrawerOpen = false;
     this.toggleDrawer = Action.create(() => isDrawerOpen = !isDrawerOpen, false);
     this.toggleDrawer.result.toProperty(this, 'isDrawerOpen');
+
+    when(listViewModel, x => x.selectedChannel)
+      .toProperty(this, 'selectedChannel');
+
+    when(this, x => x.selectedChannel)
+      .filter(c => !!c && isChannel(c))
+      .switchMap(c => {
+        this.store.channels.invalidate(c.id);
```

Comment:
Since `channels` was seeded with data from `users.counts`, we have to `invalidate`. 😞